pkg/collector/corechecks/snmp/snmp.go

```diff
@@ -0,0 +1,107 @@
+package snmp
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	snmpCheckName = "snmp"
+)
+
+// Check aggregates metrics from one Check instance
+type Check struct {
+	core.CheckBase
+	config  snmpConfig
+	session sessionAPI
+	sender  metricSender
+}
+
+// Run executes the check
+func (c *Check) Run() error {
+	sender, err := aggregator.GetSender(c.ID())
+	if err != nil {
+		return err
+	}
+
+	tags := []string{"snmp_device:" + c.config.IPAddress}
+	tags = append(tags, c.config.Tags...)
+
+	c.sender = metricSender{sender}
+
+	log.Infof("c.config.Metrics: %#v\n", c.config.Metrics) // TODO: remove me
+
+	// Create connection
+	err = c.session.Connect()
+	if err != nil {
+		log.Errorf("Connect() err: %v", err)
+	}
+	defer c.session.Close() // TODO: handle error?
+
+	snmpValues, err := c.fetchValues(err)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("scalarValues: %#v\n", snmpValues.scalarValues) // TODO: remove me
+	log.Infof("columnValues: %#v\n", snmpValues.columnValues) // TODO: remove me
+
+	// Report metrics
+	c.sender.reportMetrics(c.config.Metrics, c.config.MetricTags, snmpValues, tags)
```

Comment:
As written in `reportMetrics`, I would split it in `enrichMetrics` to add metric tags + `sendMetrics`

Reply:
Addressed here: https://github.com/DataDog/datadog-agent/pull/6749#discussion_r537356483

Reply:
I'm closing this discussion, but feel free to reopen it if needed.